File: ArticuloApp.Library/src/Solicitud.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 *
 * @author Administrator
 */
public class Solicitud {
    private String codigo;
    private Enumeraciones.TiposSolicitud tipo;
    private Articulo articuloEmisor;
    private Articulo articuloReceptor;
    private Enumeraciones.EstadosSolicitud estado;

    public String getCodigo() {
        return codigo;
    }

    public void setCodigo(String codigo) {
        this.codigo = codigo;
    }

    public Enumeraciones.TiposSolicitud getTipo() {
        return tipo;
    }

    public void setTipo(Enumeraciones.TiposSolicitud tipo) {
        this.tipo = tipo;
    }

    public Articulo getArticuloEmisor() {
        return articuloEmisor;
    }

    public void setArticuloEmisor(Articulo articuloEmisor) {
        this.articuloEmisor = articuloEmisor;
    }

    public Articulo getArticuloReceptor() {
        return articuloReceptor;
    }

    public void setArticuloReceptor(Articulo articuloReceptor) {
        this.articuloReceptor = articuloReceptor;
    }

    public Enumeraciones.EstadosSolicitud getEstado() {
        return estado;
    }

    public void setEstado(Enumeraciones.EstadosSolicitud estado) {
        this.estado = estado;
    }
}

File: ArticuloApp/ArticuloApp.Entidades/Solicitud.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ArticuloApp.Entidades
{
    public class Solicitud
    {
        public int id { get; set; }
        public Enumeraciones.TiposSolicitud tipo { get; set; }
        public Articulo articuloEmisor { get; set; }
        public Articulo articuloReceptor { get; set; }
        public Enumeraciones.EstadosSolicitud estado { get; set; }

        public Solicitud(Enumeraciones.TiposSolicitud tipo, Articulo articuloEmisor, Articulo articuloReceptor, Enumeraciones.EstadosSolicitud estado)
        {
            this.id = default(int);
            this.tipo = tipo;
            this.articuloEmisor = articuloEmisor;
            this.articuloReceptor = articuloReceptor;
            this.estado = estado;
        }

        public Solicitud()
        {
            this.id = default(int);
            this.tipo = Enumeraciones.TiposSolicitud.Intercambio;
            this.articuloEmisor = new Articulo();
            this.articuloReceptor = new Articulo();
            this.estado = Enumeraciones.EstadosSolicitud.PorResponder;
        }
    }
}
